### ISSUE TITLE
Pod identity in ztunnel could be mistaken

### ISSUE THREAD

**Issue by linsun** [istio/istio#46628]
**Describe the feature request**

Sorry if this is a dup but I can't find another issue related to this.  In the very small chance scenario, it is possible pod identity in ztunnel could be mistaken when ztunnel isn't getting update from k8s API server and using a stale IP from workload xDS, as described in the blog:  https://www.solo.io/blog/could-network-cache-based-identity-be-mistaken/

This could be considered a beta blocker.  I can update the test scripts from the blog if needed to demonstrate this problem.

cc @kdorosh @bleggett to chime in additional details.

**Affected product area (please put an X in all that apply)**

[ x ] Ambient
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by keithmattix** [comment]
In the ztunnel case, this could only happen if old workload pointed to by the stale IP happened to be on the same node as the ztunnel in question. The probability is smaller, but it could probably still happen. I feel like to avoid this, ztunnel would have to get access to and utilize the service account token for the workloads. Either that or enforce a harsh ttl on the ip address xDS

**Comment by bleggett** [comment]
IPs are not identities. They're addresses. As long as we treat them as identities the potential for this kind of problem exists.

**Comment by keithmattix** [comment]
Don't we use the IP to determine which identity ztunnel impersonates? That effectively makes it a proxy for identity

**Comment by bleggett** [comment]
> Don't we use the IP to determine which identity ztunnel impersonates? That effectively makes it a proxy for identity


Yep.

And we don't check or validate the identity of the workload with any other factor, which makes it something of a weak point.

(Though less of a weak point than other kinds of single-factor identity attestation like passwords or preshaped keys, because IPs are values that have external guarantees up to a point).

**Comment by hzxuzhonghu** [comment]
We can mitigate the mistake in some degree via checking the workload must reside in the same node as ztunnel:

For outbound: checking the original source workload, can not validate whether the dst workload is stale or not.

For inbound: checking the original dst workload(maybe during tls handshake), cannot validate the original source workload

**Comment by bleggett** [comment]
If workload certs were guaranteed unique per-pod, this would be potentially easier to catch in-mesh with timely workload identity invalidation/revocation and crosschecking.

Both of those are likely out of the question tho, so yeah we're at the mercy of the xDS cache when it comes to matching IP<->identity in ztunnel.

**Comment by howardjohn** [comment]
Initial idea was IP address was temporary and we would use something unique to the pod (likely network device (veth)) longer term. Obviously we haven't implemented that

**Comment by linsun** [comment]
An update from on this problem based on LinkedIn discussion:

With ztunnel, the problem easily solvable, We (at [solo.io](http://solo.io/)) are working to upstream a better model that solves this problem entirely by accepting/originating ztunnel connections from within the pod's netns. Stay tuned!

cc @bleggett @yuval-k

**Comment by linsun** [comment]
A proposal has been presented to upstream today - In-pod Traffic tradirection model proposal (ambient that works on any CNI)
https://docs.google.com/document/d/1dynKlnNgIOywv3cwuw_2RCk_SKFRs7IrESaC8_r-sm0

**Comment by yuval-k** [comment]
once in-pod mode lands, this becomes a lot easier to do

**Comment by yuval-k** [comment]
p.o.c branch here https://github.com/yuval-k/ztunnel/tree/inpod-dest-id (will need corresponding changes in istiod)

**Comment by howardjohn** [comment]
The ztunnel side is complete, but we still need to implement the istio CNI side
